pkg/token/service.go

```diff
@@ -56,27 +54,30 @@ type RefreshTokenData struct {
 }
 
 type tokenService struct {
-	repository                    repository
-	privateKey                    *rsa.PrivateKey
-	publicKey                     *rsa.PublicKey
-	accessTokenExpirationSeconds  int
-	refreshTokenSecretKey         string
-	refreshTokenExpirationSeconds int
+	repository     repository
+	privateKey     *rsa.PrivateKey
+	publicKey      *rsa.PublicKey
+	authentication config.Authentication
 }
 
-func (t tokenService) GetTokens(user *model.User, previousRefreshTokenId string) (*Tokens, error) {
+func (t tokenService) GetTokens(user *model.User, previousRefreshTokenId string, rememberMe bool) (*Tokens, error) {
 	if previousRefreshTokenId != "" {
 		if err := t.repository.DeleteRefreshToken(user.ID, previousRefreshTokenId); err != nil {
 			return nil, errdef.NewUnauthorized("could not delete previous refreshToken for user.Id: %d, tokenId: %s", user.ID, previousRefreshTokenId)
 		}
 	}
 
-	accessToken, err := helper.GenerateAccessToken(user, t.privateKey, t.accessTokenExpirationSeconds)
+	accessToken, err := helper.GenerateAccessToken(user, t.privateKey, t.authentication.AccessTokenExpirationSeconds)
```

Comment:
I think our code would benefit from using `time.Duration` for such values. We could stick to the `_SECONDS` env vars but then translate into the `time.Duration` type when these configs leave the config package. `time.Duration` is easier to work with when computing things like a deadline, ... and since its typed we can shorten our var names 😋

Reply:
As much as I agree I'd say it's beyond the scope of this PR to change our current pattern